File: employee-management-app/src/app/pages/profile/models/dto/createUserProject.ts
import {UserProfile} from "../userProfile";
import {Project} from "../project";

export class CreateUserProject {
  engineerId: string;
  projectId: string;
  startDate: Date;
  endDate: Date;
  description: string;

  constructor(
              user: string,
              project: string,
              startDate: Date,
              endDate: Date,
              description: string
  ) {
    this.engineerId = user;
    this.projectId = project;
    this.startDate = startDate;
    this.endDate = endDate;
    this.description = description;
  }
}

File: employee-management/src/main/java/com/dreamteam/employeemanagement/dto/profile/CreateUserProject.java
package com.dreamteam.employeemanagement.dto.profile;

import com.dreamteam.employeemanagement.model.Account;
import com.dreamteam.employeemanagement.model.Project;
import jakarta.persistence.Column;
import jakarta.persistence.ManyToOne;

import java.time.LocalDateTime;
import java.util.UUID;

public class CreateUserProject {
    private int engineerId;
    private int projectId;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private String description;
}
